google-cloud-bigtable/lib/google/cloud/bigtable/rows_reader.rb

```diff
@@ -170,13 +189,13 @@ def retryable?
         # @return [Boolean]
         #
         def start_key_read? range
-          start_key = if range.start_key_closed.empty?
-                        range.start_key_open
-                      else
-                        range.start_key_closed
-                      end
-
-          start_key.empty? || last_key >= start_key
+          if range.start_key_closed.empty? && range.end_key_open.empty?
+            true
+          elsif !range.start_key_closed.empty?
+            last_key >= range.start_key_closed
+          else
+            last_key > range.start_key_open
+          end
```

Comment:
I think it would cleaner to structure this as:
```
if !range.start_key_closed.empty?
    last_key >= range.start_key_closed
elseif !range.start_key_open.empty?
  last_key > range.start_key_closed
else
  # start is unbounded
  true
end
```